vmdk_plugin/utils/plugin_utils/plugin_utils.go

```diff
@@ -70,27 +70,54 @@ func AlreadyMounted(name string, mountRoot string) bool {
 
 // GetDatastore - get datastore from volume metadata
 // Note "datastore" key is defined in vmdkops service
-func GetDatastore(name string, d drivers.VolumeDriver) (string, error) {
+func GetDatastore(name string, d drivers.VolumeDriver) (string, map[string]interface{}, error) {
 	volumeMeta, err := d.GetVolume(name)
 	if err != nil {
 		log.Errorf("Unable to get volume metadata %s (err: %v)", name, err)
-		return "", err
+		return "", nil, err
 	}
-	return volumeMeta["datastore"].(string), nil
+	return volumeMeta["datastore"].(string), volumeMeta, nil
 }
 
-// GetFullVolumeName - append datastore to the volume name
-func GetFullVolumeName(name string, datastoreName string, d drivers.VolumeDriver) (string, error) {
+// GetNameFromRefmap - get names from refmap
```

Comment:
nit: please add comment. Something like  "finds an entry for volume. Entry can be in vol@ds format, Volums is passed in as a short name, with no '@ds' (btw 0 is this what you intended to do ? )